### PR TITLE
Created Decedent model with repository

### PR DESCRIPTION
**Pull request by brzezin0**

### CODE REVIEW

File: src/main/java/com/app/backend/model/Decedent.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name = "_decedent")
+public class Decedent {
+
+    @Id
+    @GeneratedValue
+    private Integer id;
+    @Column(unique = true, nullable = false)
```
Comment (Ikar17) [inline src/main/java/com/app/backend/model/Decedent.java]: name is not necessary to be unique

File: src/main/java/com/app/backend/model/Decedent.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    private String name;
+    private String surname;
+    @Column(nullable = false)
```
Comment (Ikar17) [inline src/main/java/com/app/backend/model/Decedent.java:23]: Date of birth can be null

File: src/main/java/com/app/backend/repository/DecedentRepository.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.app.backend.repository;
+
+import com.app.backend.model.Decedent;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public interface DecedentRepository extends JpaRepository<Decedent, Integer> {
```
Comment (WiktoriaKrzyzanska) [inline src/main/java/com/app/backend/repository/DecedentRepository.java:10]: Do we need to use the JpaRepository? JpaRepository is an extension of the CRUDRepository but it’s used when we want to perform operations on a batch.

File: src/main/java/com/app/backend/model/Decedent.java
Comment (WiktoriaKrzyzanska) [inline src/main/java/com/app/backend/model/Decedent.java:1]: While we don’t have strict rules regarding naming model is something what stands for POJO and for stuff related to databases we should use entity

File: src/main/java/com/app/backend/model/Decedent.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    @Lob
+    @Column(name = "tombstone_image")
+    private byte[] tombstoneImage;
+
+    @ManyToOne
+    @JoinColumn(name="user_id")
+    private User user;
+
+    public int calculateAge(){
```
Comment (WiktoriaKrzyzanska) [inline src/main/java/com/app/backend/model/Decedent.java]: You can add here //todo for example as we don’t use this method so it’s good to add todo.

File: src/main/java/com/app/backend/model/Decedent.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        if(birthDate!=null&&deathDate!=null){
+            return Period.between(birthDate,deathDate).getYears();
+        }else{
+            return 0;
+        }
+    }
+
```
Comment (WiktoriaKrzyzanska) [inline src/main/java/com/app/backend/model/Decedent.java:52]: You can formate the file using ctrl alt shift l. It’s a good practice

**Review by WiktoriaKrzyzanska — APPROVED** [review]

File: src/main/java/com/app/backend/model/Decedent.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
+package com.app.backend.model;
```
Comment (WiktoriaKrzyzanska) [inline src/main/java/com/app/backend/model/Decedent.java:1]: abcd

**Review by WiktoriaKrzyzanska — CHANGES_REQUESTED** [review]

**Review by Ikar17 — APPROVED** [review]

**Review by WiktoriaKrzyzanska — APPROVED** [review]

**Pull request merged**